plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererFileCrawler.kt

```diff
@@ -86,14 +90,40 @@ abstract class CodeWhispererFileCrawler : FileCrawler {
     }.orEmpty()
 
     override fun listCrossFileCandidate(target: PsiFile): List<VirtualFile> {
+        val candidates = if (CodeWhispererFeatureConfigService.getInstance().getCrossfileConfig()) {
+            val previousSelected: List<VirtualFile> = listPreviousSelectedFile(target)
+                .filter { !isTestFile(it, target.project) && isSameDialect(it.extension) }
+            val neighbors: List<VirtualFile> = neighborFiles(target, NEIGHBOR_FILES_DISTANCE)
+                .filter { !isTestFile(it, target.project) && isSameDialect(it.extension) }
+            val openedFiles = listAllOpenedFilesSortedByDist(target)
+                .filter { !isTestFile(it, target.project) && isSameDialect(it.extension) }
+
+            val result = previousSelected.take(3) + neighbors + openedFiles
```

Comment:
>let’s sort like this.
take first 3 files from MRU (open file), then neighbor files with distance=0, then 1 , then 2, then the rest open files.

defined by science